### PR TITLE
[Fix](Variant) support materialize view for variant and accessing variant subcolumns

### PR DESCRIPTION
**Pull request by eldenmoon**
1. fix schema change with path lost and lead to invalid data read
2. support element_at function in BE side and use simdjson to parse data
3. fix multi slot expression

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/functions/function_variant_element.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+private:
+    static Status get_element_column(const ColumnObject& src, const ColumnPtr& index_column,
+                                     ColumnPtr* result) {
+        std::string field_name = index_column->get_data_at(0).to_string();
+        if (src.empty()) {
+            *result = ColumnObject::create(true);
+            return Status::OK();
+        }
+        if (src.is_scalar_variant() &&
+            WhichDataType(remove_nullable(src.get_root_type())).is_string_or_fixed_string()) {
```
Comment (xiaokang) [inline be/src/vec/functions/function_variant_element.cpp:108]: variant column should be string?

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 2]
```diff
@@ -446,11 +446,12 @@ Status Segment::new_column_iterator_with_path(const TabletColumn& tablet_column,
             return Status::OK();
         }
         bool output_as_raw_json = true;
+        // Not use path info in TabletColumn since the path info may not be set in schema change process
+        vectorized::PathInData path(tablet_column.name_lower_case());
         // Alter table operation should read the whole variant column, since it does not aware of
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment.cpp:451]: use root_path

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 3]
```diff
@@ -1658,27 +1658,30 @@ && findOlapScanNodesByPassExchangeAndJoinNode(inputFragment.getPlanRoot())) {
     }
 
     // Get top most PushDownToProjectionFunction from expression
-    private Expression getOriginalFunctionForRewritten(NamedExpression expression) {
-        List<Expression> targetExpr = expression.collectFirst(PushDownToProjectionFunction.class::isInstance);
-        if (!targetExpr.isEmpty()) {
-            return targetExpr.get(0);
-        }
-        return null;
+    private List<Expression> getPushDownToProjectionFunctionForRewritten(NamedExpression expression) {
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1661]: explained

File: be/src/olap/rowset/segment_v2/segment.cpp (hunk 2 above)
Comment (eldenmoon) [reply be/src/olap/rowset/segment_v2/segment.cpp:449]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**